src/windows/simulate.rs

```diff
@@ -58,69 +76,54 @@ fn keyboard_event(flags: u32, vk: u16, scan: u16) -> Result<(), SimulateError> {
-            Button::Right => mouse_event(MOUSEEVENTF_RIGHTDOWN, 0, 0, 0),
-            Button::Unknown(code) => mouse_event(MOUSEEVENTF_XDOWN, 0, 0, *code as i32),
+            Button::Left => sim_mouse_event(MOUSEEVENTF_LEFTDOWN, 0, 0, 0),
+            Button::Middle => sim_mouse_event(MOUSEEVENTF_MIDDLEDOWN, 0, 0, 0),
+            Button::Right => sim_mouse_event(MOUSEEVENTF_RIGHTDOWN, 0, 0, 0),
+            Button::Unknown(code) => sim_mouse_event(MOUSEEVENTF_XDOWN, 0, 0, (*code).into()),
         },
         EventType::ButtonRelease(button) => match button {
-            Button::Left => mouse_event(MOUSEEVENTF_LEFTUP, 0, 0, 0),
-            Button::Middle => mouse_event(MOUSEEVENTF_MIDDLEUP, 0, 0, 0),
-            Button::Right => mouse_event(MOUSEEVENTF_RIGHTUP, 0, 0, 0),
-            Button::Unknown(code) => mouse_event(MOUSEEVENTF_XUP, 0, 0, *code as i32),
+            Button::Left => sim_mouse_event(MOUSEEVENTF_LEFTUP, 0, 0, 0),
+            Button::Middle => sim_mouse_event(MOUSEEVENTF_MIDDLEUP, 0, 0, 0),
+            Button::Right => sim_mouse_event(MOUSEEVENTF_RIGHTUP, 0, 0, 0),
+            Button::Unknown(code) => sim_mouse_event(MOUSEEVENTF_XUP, 0, 0, (*code).into()),
         },
         EventType::Wheel { delta_x, delta_y } => {
-            let result_x = if *delta_x != 0 {
-                mouse_event(
+            if *delta_x != 0 {
+                sim_mouse_event(
                     MOUSEEVENTF_HWHEEL,
+                    (c_short::try_from(*delta_x).map_err(|_| SimulateError)? * WHEEL_DELTA) as u32,
                     0,
-                    (delta_x * WHEEL_DELTA as i64) as i32,
                     0,
-                )
-            } else {
-                Ok(())
-            };
+                )?;
+            }
 
-            let result_y = if *delta_y != 0 {
-                mouse_event(
+            if *delta_y != 0 {
+                sim_mouse_event(
                     MOUSEEVENTF_WHEEL,
+                    (c_short::try_from(*delta_y).map_err(|_| SimulateError)? * WHEEL_DELTA) as u32,
                     0,
                     0,
-                    (delta_y * WHEEL_DELTA as i64) as i32,
-                )
-            } else {
-                Ok(())
-            };
-
-            if result_x.is_ok() && result_y.is_ok() {
-                Ok(())
-            } else {
-                Err(SimulateError)
+                )?;
             }
+            Ok(())
         }
         EventType::MouseMove { x, y } => {
             let width = unsafe { GetSystemMetrics(SM_CXVIRTUALSCREEN) };
             let height = unsafe { GetSystemMetrics(SM_CYVIRTUALSCREEN) };
 
-            mouse_event(
-                MOUSEEVENTF_MOVE | MOUSEEVENTF_ABSOLUTE,
+            sim_mouse_event(
+                MOUSEEVENTF_MOVE | MOUSEEVENTF_ABSOLUTE | MOUSEEVENTF_VIRTUALDESK,
                 0,
-                (*x * 65335.0) as i32 / width,
-                (*y * 65335.0) as i32 / height,
```

Comment:
typo! 65335 looks a lot like 65535. This took me longer to find than I'd like to admit.